src/sqs.rs

```diff
@@ -185,24 +188,34 @@ mod test_send_messages_concurrently {
             .queue_url
             .unwrap();
 
-        // FIXME: use batch operations
         let mut results: Vec<usize> = vec![];
         while let Ok(x) = client
             .receive_message()
+            .max_number_of_messages(i32::try_from(MAX_MESSAGES).ok().unwrap())
             .wait_time_seconds(1)
             .queue_url(&queue_url)
             .send()
             .await
         {
             match x.messages {
                 Some(ref messages) => {
-                    assert_eq!(messages.len(), 1);
-                    let message = &messages[0];
-                    results.push(message.body.as_ref().unwrap().parse().unwrap());
+                    assert!(messages.len() <= MAX_MESSAGES);
+
+                    let results_delete = messages
+                        .iter()
+                        .map(|msg| {
+                            results.push(msg.body.as_ref().unwrap().parse().unwrap());
+                            DeleteMessageBatchRequestEntry::builder()
+                                .receipt_handle(msg.receipt_handle.as_ref().unwrap())
+                                .set_id(Some(msg.message_id.as_ref().unwrap().parse().unwrap()))
+                                .build()
+                        })
+                        .collect::<Vec<_>>();
```

Comment:
This all looks good, but I have a fun extra-credit challenge for you 😁

I believe we can remove most of the uses of references here by consuming `messages` by value. Broadly:

* In the match arm on line `201`, remove the `ref` so that its just matching on `Some(messages)`. This will cause the value in `messages` to be an owned `Vec<Message>` rather than a reference `&Vec<Message>`.
* When iterating over `messages`, use `.into_iter()` rather than `.iter()` on line `205`. This will cause the value of `msg` passed to each iteration of the `map` closure to be an owned `Message` rather than a reference `&Message`.
* When parsing the result on line `207`, you'll no longer need to use `as_ref()`. This is the only place you want to use `msg.body`, so you can take ownership of it using `unwrap`, parse it, and then allow it to be dropped when the parsing is done.
* When setting the receipt handle on line `209`, the current code uses `as_ref()` to pass a reference to the receipt-handle string, so the called method will need to take its own internal copy of the passed-by-reference string. Instead, you can now call `set_receipt_handle(msg.receipt_handle)` to take the owned `Option<String>` value out of `msg` and pass it directly into the method, avoiding the need for it to take a copy.
* Likewise on line `210`, you can now call `set_id(msg.message_id)` to take the owned `Option<String>` out of `msg.message_id` and pass it directory into the method, avoiding the need for a copy.

The performance improvement from this change will be absolutely minimal in practice (we just avoid copying a few short strings) but I think walking through the change would be a really useful exercise to get familiar with Rust's ownership model and the way you can pass owned values to avoid copying data.